src/main/java/org/apache/accumulo/testing/continuous/ThreadedContinousInputFormat.java

```diff
@@ -0,0 +1,237 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.accumulo.testing.continuous;
+
+import static org.apache.accumulo.testing.continuous.ContinuousIngest.*;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.security.SecureRandom;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.concurrent.*;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.LongAdder;
+import java.util.stream.IntStream;
+import java.util.zip.CRC32;
+import java.util.zip.Checksum;
+
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.security.ColumnVisibility;
+import org.apache.accumulo.testing.TestProps;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.mapreduce.*;
+
+import com.google.common.collect.Maps;
+
+/**
+ * Purpose and Justification: Defines an asynchronous input format that parallelizes the creation of
+ * the data. This aims to move the bar from a data generation + I/O ( spill ) problem to the right
+ * so that if we are able to optimize the spill path we ultimately minimize the impact of data
+ * generation.
+ *
+ * Note that this class leverages BulkKey versus Key. This was not merged into ContinousInputformat
```

Comment:
I was very cautious to keep the existing functionality. Adding threads to add threads doesn't always benefit so I wanted to isolate this functionality; however, I made a change while submitting this PR for ContinuousInputFormat to use BulkKey vs Key, so I will at the very least extend ContinuousInputFormat in ThreadedContinousInputFormat